exo-deploy/src/deployers/aws-deployer.ls

```diff
@@ -39,20 +41,62 @@ class AwsDeployer
 
   deploy: ->
     @terraform
-      ..get (err) ->
+      ..get (err) ~>
         | err => return process.stdout.write err.message
-        process.stdout.write "terraform starting deploy to AWS"
-        ..apply (err) ->
-          | err => return process.stdout.write err.message
+        @_get-hosted-zone-id destroy: no, (err, hosted-zone-id) ~>
+          | err => process.stdout.write "Cannot get hosted zone id #{err.message}" ; return
+          ..apply {hosted-zone-id}, (err) ->
+            | err => return process.stdout.write err.message
 
 
   nuke: ->
```

Comment:
It's better practice to have async methods always accept a `done` parameter and call it at the end. Otherwise the method looks like it is synchronous, but it isn't. It also helps use the method for flexible in different places.